src/main/java/redis/clients/jedis/DefaultJedisSocketFactory.java

```diff
@@ -46,7 +51,7 @@ public DefaultJedisSocketFactory(String host, int port, int connectionTimeout, i
   }
 
   public DefaultJedisSocketFactory(HostAndPort hostAndPort, JedisClientConfig config) {
-    this.hostAndPort = hostAndPort;
+    this.hostAndPort.set(hostAndPort);
```

Comment:
Shouldn't we check null here?
It might be better to keep the default HostAndPort.

Reply:
For consistency with the rest of the Jedis code I would opt for not null checking here, all the constructors I have seen in the codebase do accept null (and while it would be a bad idea) it is only possible if the user does that on purpose.

I don't mind adding it though if you feel strongly about it

Reply:
I'm asking because of this https://github.com/redis/jedis/pull/2293/files#diff-5c1bd2b61d33731ac162212486900a5408ba527baddcdb5decdf41d1e4f26c53R37 

IMO, we should either check null here or avoid sending null there (`this(null, config);`).

Reply:
Ah, I think I misunderstood you, I though you meant `if (hostAndPort == null) throw new ArgumentException("HostAndPort cannot be null"` but now I think you meant `if (hostAndPort != null) this.hostAndPort.set(hostAndPort)`?

If it's the latter then I totally agree and I think it makes perfect sense, is that what you meant?

Reply:
YES